subsys/bootloader/bl_crypto/Kconfig

```diff
@@ -90,29 +96,40 @@ choice SB_CRYPTO_HASH
 	help
 	  Crypto implementation to use for hash generation in Bootloader.
 
-	config SB_CRYPTO_OBERON_SHA256
-		bool "Software SHA256"
-		select SB_SHA256
-		help
-		  Software implementation of SHA256.
-
-	config SB_CRYPTO_CC310_SHA256
-		bool "Hardware SHA256" if HAS_HW_NRF_CC310
-		select NRF_CC310_BL
-		select SB_SHA256
-		help
-		  Hardware implementation of SHA256.
-
-	config SB_CRYPTO_CLIENT_SHA256
-		bool "Use another image's SHA256 implementation"
-		select BL_SHA256_EXT_API_ATLEAST_REQUIRED
-		select SB_SHA256
-		help
-		  Using EXT_APIs from fw_info.
-
-	config SB_CRYPTO_NO_SHA256
-		bool "Disable SHA256 support"
-		select SB_SHA256
+config SB_CRYPTO_OBERON_SHA256
+	bool "Software SHA256"
+	select SB_SHA256
+	help
+	  Software implementation of SHA256.
+
+config SB_CRYPTO_CC310_SHA256
+	bool "Hardware SHA256" if HAS_HW_NRF_CC310
+	select NRF_CC310_BL
+	select SB_SHA256
+	help
+	  Hardware implementation of SHA256.
+
+config SB_CRYPTO_CLIENT_SHA256
+	bool "Use another image's SHA256 implementation"
+	select BL_SHA256_EXT_API_ATLEAST_REQUIRED
+	select SB_SHA256
+	help
+	  Using EXT_APIs from fw_info.
+
+config SB_CRYPTO_NO_SHA256
+	bool "Disable SHA256 support"
+	select SB_SHA256
+
+config SB_CRYPTO_PSA_SHA512
+	bool "PSA SHA512 support"
+	depends on NRF_SECURITY
+	select SB_SHA512
+	select PSA_WANT_ALG_SHA_512
+	select PSA_WANT_ALG_PURE_EDDSA
+	select PSA_WANT_ECC_TWISTED_EDWARDS_255
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT
```

Comment:
I think this part is only needed for ED25519 and SHA512 itself does not require.